chia/wallet/trade_manager.py

```diff
@@ -409,32 +411,39 @@ async def respond_to_offer(self, offer: Offer, fee=uint64(0)) -> Tuple[bool, Opt
                             memos=[],
                         )
                     )
+                else:  # This is change
+                    addition_dict.setdefault(wallet_id, [])
+                    addition_dict[wallet_id].append(addition)
 
         # While we want additions to show up as separate records, removals of the same wallet should show as one
         removal_dict: Dict[uint32, List[Coin]] = {}
         for removal in removals:
-            if removal.name() in settlement_parents:
-                wallet_info = await self.wallet_state_manager.get_wallet_id_for_puzzle_hash(removal.puzzle_hash)
-                if wallet_info is not None:
-                    wallet_id, _ = wallet_info
-                    removal_dict.setdefault(wallet_id, [])
-                    removal_dict[wallet_id].append(removal)
+            wallet_info = await self.wallet_state_manager.get_wallet_id_for_puzzle_hash(removal.puzzle_hash)
+            if wallet_info is not None:
+                wallet_id, _ = wallet_info
+                removal_dict.setdefault(wallet_id, [])
+                removal_dict[wallet_id].append(removal)
 
         for wid, grouped_removals in removal_dict.items():
             wallet = self.wallet_state_manager.wallets[wid]
-            to_puzzle_hash = bytes32([0] * 32)  # We use all zeros to be clear not to send here
+            to_puzzle_hash = bytes32([1] * 32)  # We use all zeros to be clear not to send here
             removal_tree_hash = Program.to([rem.as_list() for rem in grouped_removals]).get_tree_hash()
+            # We also need to calculate the sent amount
+            removed: int = sum(c.amount for c in grouped_removals)
+            change_coins: List[Coin] = addition_dict[wid] if wid in addition_dict else []
+            change_amount: int = sum(c.amount for c in change_coins)
+            sent_amount: int = removed - change_amount
```

Comment:
I'm probably missing something basic here, but I'm not clear on why `change_amount` is being subtracted from `removal`. Isn't change_amount coming from the additions?

I was initially looking at this wondering if the result could be negative, because `sent_amount` is then stuffed into a uint64 when creating the TransactionRecord.